src/transformers/models/vision_encoder_decoder/modeling_flax_vision_encoder_decoder.py

```diff
@@ -0,0 +1,849 @@
+
+            >>> pixel_values = feature_extractor(images=image, return_tensors="np").pixel_values
+
+            >>> # use GPT2's eos_token as the pad as well as eos token
+            >>> model.config.eos_token_id = model.config.decoder.eos_token_id
+            >>> model.config.pad_token_id = model.config.eos_token_id
+
+            >>> # generation
+            >>> sequences = model.generate(pixel_values, num_beams=4, max_length=12).sequences
+
+            >>> captions = tokenizer_output.batch_decode(sequences, skip_special_tokens=True)
+        """
+
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        # prepare encoder inputs
+
+        # `FlaxViTModel` expects channel first format, but `FlaxViTModule` expects channel last format.
+        # Currently, we assume this holds for all Flax vision models, and perform a transpose here.
+        pixel_values = jnp.transpose(pixel_values, (0, 2, 3, 1))
+
+        # prepare decoder inputs
+        if decoder_input_ids is None:
+            raise ValueError("`decoder_input_ids` can't be `None`.")
+        if decoder_attention_mask is None:
+            decoder_attention_mask = jnp.ones_like(decoder_input_ids)
+        if decoder_position_ids is None:
+            batch_size, sequence_length = decoder_input_ids.shape
+            decoder_position_ids = jnp.broadcast_to(
+                jnp.arange(sequence_length)[None, :], (batch_size, sequence_length)
+            )
+
+        # vision models don't use attention mask - set to `None` as done in `VisionEncoderDecoderModel`
+        encoder_attention_mask = None
+
+        # Handle any PRNG if needed
+        rngs = {"dropout": dropout_rng} if dropout_rng is not None else {}
+
+        return self.module.apply(
+            {"params": params or self.params},
+            pixel_values=jnp.array(pixel_values, dtype=self.dtype),
+            attention_mask=encoder_attention_mask,
+            decoder_input_ids=jnp.array(decoder_input_ids, dtype="i4"),
+            decoder_attention_mask=jnp.array(decoder_attention_mask, dtype="i4"),
+            decoder_position_ids=jnp.array(decoder_position_ids, dtype="i4"),
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            deterministic=not train,
+            rngs=rngs,
+        )
+
+    def prepare_inputs_for_generation(
+        self,
+        decoder_input_ids,
+        max_length,
+        attention_mask: Optional[jnp.DeviceArray] = None,
+        decoder_attention_mask: Optional[jnp.DeviceArray] = None,
+        encoder_outputs=None,
+        **kwargs
```

Comment:
is this needed? If we need it for the attention_mask, then let's leave a comment:

`# kwargs is added at the moment to "absorb" the useless `attention_mask` argument